### PR TITLE
Add Glue table name to outputs, minor fix for optional KMS key PR

### PR DESCRIPTION
**Pull request by sbp-mbaan**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -1,6 +1,7 @@
 locals {
-  kms_policy     = var.kms_key_arn != null ? data.aws_iam_policy_document.firehose_kms_role[0].json : null
-  parquet_prefix = "year=!{timestamp:yyyy}/month=!{timestamp:MM}/day=!{timestamp:dd}/hour=!{timestamp:HH}/"
+  kinesis_source_configuration = var.kinesis_arn != null ? { create = true } : {}
```
Comment (svanharmelen) [inline main.tf:2]:
```suggestion
  kinesis = var.kinesis_arn != null ? { create = true } : {}
```

File: outputs.tf [hunk 2]
```diff
@@ -3,6 +3,11 @@ output "arn" {
   description = "ARN of the firehose delivery stream"
 }
 
+output "glue_table_name" {
+  description = "Name of the Glue Table"
+  value       = replace(var.name, "-", "_")
```
Comment (svanharmelen) [inline outputs.tf:8]:
```suggestion
  value       = aws_glue_catalog_table.default.name
  description = "Name of the Glue Table"
```

**Review by svanharmelen — CHANGES_REQUESTED** [review]

File: main.tf (hunk 1 above)
Comment (svanharmelen) [reply main.tf:2]: Let's try to keep it a bit concise 😏

File: outputs.tf (hunk 2 above)
Comment (svanharmelen) [reply outputs.tf:8]: Check the ordering and let's use the actual reference instead of repeating some logic here...

Comment (sbp-mbaan) [reply outputs.tf]: I thought I read it did not return the name, only an ARN. But that was wrong. Replaced.

File: main.tf (hunk 1 above)
Comment (sbp-mbaan) [reply main.tf]: Fair.

File: main.tf [hunk 3]
```diff
@@ -1,6 +1,7 @@
 locals {
-  kms_policy     = var.kms_key_arn != null ? data.aws_iam_policy_document.firehose_kms_role[0].json : null
-  parquet_prefix = "year=!{timestamp:yyyy}/month=!{timestamp:MM}/day=!{timestamp:dd}/hour=!{timestamp:HH}/"
+  kinesis = var.kinesis_arn != null ? { create = true } : {}
```
Comment (svanharmelen) [inline main.tf:2]: Having another look, why not something like this:

```suggestion
  kinesis = var.kinesis_arn != null ? [var.kinesis_arn] : []
```

File: main.tf [hunk 4]
```diff
@@ -1,6 +1,7 @@
...
+  kms_policy                   = var.kms_key_arn != null ? data.aws_iam_policy_document.firehose_kms_role[0].json : null
+  parquet_prefix               = "year=!{timestamp:yyyy}/month=!{timestamp:MM}/day=!{timestamp:dd}/hour=!{timestamp:HH}/"
```
Comment (svanharmelen) [inline main.tf:4]: You need to run `terraform fmt` again now...

File: main.tf [hunk 5]
```diff
@@ -94,10 +95,10 @@ resource "aws_kinesis_firehose_delivery_stream" "default" {
   tags        = var.tags
 
   dynamic "kinesis_source_configuration" {
-    for_each = toset([var.kinesis_arn])
+    for_each = local.kinesis
```
Comment (svanharmelen) [inline main.tf:98]:
```suggestion
    for_each = toset(local.kinesis)
```

File: main.tf [hunk 6]
```diff
@@ -94,10 +95,10 @@ resource "aws_kinesis_firehose_delivery_stream" "default" {
...
 
     content {
-      kinesis_stream_arn = each.value
+      kinesis_stream_arn = var.kinesis_arn
```
Comment (svanharmelen) [inline main.tf:101]:
```suggestion
      kinesis_stream_arn = each.value
```

**Review by svanharmelen — COMMENTED** [review]
Slightly updated suggestion... What do you think?

**Review by svanharmelen — APPROVED** [review]
Nice 👍🏻

**Pull request merged**